gcp/src/main/java/org/apache/iceberg/gcp/GCPProperties.java

```diff
@@ -61,6 +62,8 @@ public class GCPProperties implements Serializable {
    */
   public static final int GCS_DELETE_BATCH_SIZE_DEFAULT = 50;
 
+  private final Map<String, String> allProperties;
```

Comment:
Do we really need to keep all the properties?  The goal was to no copy all properties when we should only need the FileIO specific properties (though it looks like all properties were added to the FileIO as well which is questionable)

Reply:
this shouldn't be needed anymore once we remove `GCPProperties` from the constructors of `GCSFileIO`